hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiver.java

```diff
@@ -1582,6 +1582,36 @@ public void testPendingClusteringAfterArchiveCommit(boolean enableMetadata) thro
         "Since we have a pending clustering instant at 00000002, we should never archive any commit after 00000000");
   }
 
+  @Test
```

Comment:
For context, I had a patch on my organization's internal older build of HUDI that had the same fix as [HUDI-7207](https://issues.apache.org/jira/browse/HUDI-7207) as well as this unit test. Although [HUDI-7207](https://issues.apache.org/jira/browse/HUDI-7207) has already been identified and resolved, I wanted to open a PR for adding this unit test in case the `apache/hudi` reviewers think it is useful (and if not I can close this PR).